server/integration/testsuite/src/test/java/org/infinispan/server/test/util/ManagementClient.java

```diff
@@ -0,0 +1,221 @@
+        ops.add(Address.subsystem("datagrid-infinispan-endpoint")
+                        .and("memcached-connector", name),
+                Values.empty()
+                        .and("cache-container", cacheContainer)
+                        .and("cache", cache)
+                        .and("socket-binding", socketBinding));
+    }
+
+    public void removeMemcachedEndpoint(String name) throws Exception {
+        ops.removeIfExists(Address.subsystem("datagrid-infinispan-endpoint")
+                .and("memcached-connector", name));
+    }
+
+    public void addRestEndpoint(String name, String cacheContainer, String cache, String socketBinding) throws Exception {
+        ops.add(Address.subsystem("datagrid-infinispan-endpoint")
+                        .and("rest-connector", name),
+                Values.empty()
+                        .and("cache-container", cacheContainer)
+                        .and("cache", cache)
+                        .and("socket-binding", socketBinding));
+    }
+
+    public void removeRestEndpoint(String name) throws Exception {
+        ops.removeIfExists(Address.subsystem("datagrid-infinispan-endpoint")
+                .and("rest-connector", name));
+    }
+
+    public void addHotRodEndpoint(String name, String cacheContainer, String cache, String socketBinding) throws Exception {
+        ops.add(Address.subsystem("datagrid-infinispan-endpoint")
+                        .and("hotrod-connector", name),
+                Values.empty()
+                        .and("cache-container", cacheContainer)
+                        .and("cache", cache)
+                        .and("socket-binding", socketBinding));
+    }
+
+    public void removeHotRodEndpoint(String name) throws Exception {
+        ops.removeIfExists(Address.subsystem("datagrid-infinispan-endpoint")
+                .and("hotrod-connector", name));
+    }
+
+    public void addCacheContainer(String name, String defaultCache) throws Exception {
+        ops.add(Address.subsystem("datagrid-infinispan")
+                        .and("cache-container", name),
+                Values.empty()
+                        .and("default-cache", defaultCache)
+                        .and("statistics", "true"));
+    }
+
+    public void removeCacheContainer(String name) throws Exception {
+        ops.removeIfExists(Address.subsystem("datagrid-infinispan")
+                .and("cache-container", name));
+    }
+
+    private enum CacheTemplate {
+        DIST("distributed-cache-configuration"),
+        REPL("replicated-cache-configuration"),
+        LOCAL("local-cache-configuration");
+
+        private String type;
+
+        CacheTemplate(String type) {
+            this.type = type;
+        }
+
+        public String getType() {
+            return type;
+        }
+    }
+
+    public static class Constants {
+        public static final String NODE0_ADDRESS = System.getProperty("node0.ip", "127.0.0.1");
```

Comment:
What is the advantage of keeping those constants in separate class? Do we want to delay their initialization? If not - I would suggest moving them to `ManagementClient`.